driver-core/src/main/com/mongodb/MongoNamespace.java

```diff
@@ -57,9 +58,10 @@
     public static void checkDatabaseNameValidity(final String databaseName) {
         notNull("databaseName", databaseName);
         isTrueArgument("databaseName is not empty", !databaseName.isEmpty());
-        for (int i = 0; i < databaseName.length(); i++) {
-            isTrueArgument("databaseName does not contain '" + databaseName.charAt(i) + "'",
-                    !PROHIBITED_CHARACTERS_IN_DATABASE_NAME.contains(databaseName.charAt(i)));
+        Optional<Character> prohibitedChar = PROHIBITED_CHARACTERS_IN_DATABASE_NAME.stream()
```

Comment:
It looks like an alternative correct implementation, but I would need some convincing that it's actually more performant.  In particular, it trades a call to HashSet#contains ( O(1) )  for each character in the database name for a call to String#contains ( O(n) ) for each prohibited character.  I don't think that's a good trade.

I think better would be to keep the loop as is, but do the HashSet#contains before constructing an error string.

Reply:
I see your point. I have updated the loop.